src/main/java/uk/gov/digital/ho/hocs/audit/export/infoclient/ExportViewConstants.java

```diff
@@ -2,11 +2,22 @@
 
 public interface ExportViewConstants {
 
+        enum DATE_SECTION {DAY, MONTH, YEAR};
+        int YEAR_DIGITS = 4;
+        int MONTH_DIGITS = 2;
+        int DAY_DIGITS = 2;
+
+        String MALFORMED_DATE_REGEX = "^(\\d{4,5})\\-(0*[1-9]|0*1[12])\\-(0*0[1-9]|0*[12][0-9]|0*3[01])$";
```

Comment:
![image(2)](https://user-images.githubusercontent.com/817054/122366919-e9ff7400-cf53-11eb-91c9-686c083b16c7.png)


This is a very complex regular expression. Please either amend the Git commit message or add comments to explain what on earth is going on.

(Also, it doesn't look like it'd support October?)

Reply:
This is currently capturing the groups all the groups so do we need the day, month and year regex? You could make this a little more strict and use the capturing groups to actually give us the correct day, month and year)

Simplified regex without capturing the groups as current usage doesn't require them:
`^(?:\\d{4,5})\\-(?:0*(?:[1-9]|1[0,2]))\\-(?:0*(?:[1-9]|[12][0-9]|3[01]))$`

This modified regex also captures October.

Reply:
My comment is resolved. Leaving this open as the git commit message could probably do with more explanation/ the regex commented. The other regex's are a little complicated.